apps/gdal_rasterize_lib.cpp

```diff
@@ -50,17 +50,338 @@
 #include "ogr_api.h"
 #include "ogr_core.h"
 #include "ogr_srs_api.h"
+#include "gdalargumentparser.h"
 
 /************************************************************************/
-/*                       ArgIsNumericRasterize()                        */
+/*                          GDALRasterizeOptions()                      */
 /************************************************************************/
 
-static bool ArgIsNumericRasterize(const char *pszArg)
+struct GDALRasterizeOptions
+{
+    std::vector<int> anBandList;
+    std::vector<double> adfBurnValues;
+    bool bInverse = false;
+    std::string osFormat;
+    bool b3D = false;
+    GDALProgressFunc pfnProgress = GDALDummyProgress;
+    void *pProgressData;
+    std::vector<std::string> aosLayers;
+    std::string osSQL;
+    std::string osDialect;
+    std::string osBurnAttribute;
+    std::string osWHERE;
+    CPLStringList aosRasterizeOptions;
+    CPLStringList aosTO;
+    double dfXRes;
+    double dfYRes;
+    CPLStringList aosCreationOptions;
+    GDALDataType eOutputType;
+    std::vector<double> adfInitVals;
+    std::string osNoData;
+    OGREnvelope sEnvelop;
+    int nXSize, nYSize;
+    OGRSpatialReference oOutputSRS;
```

Comment:
Some missing initializations. I'm voluntariliy also initializing C++ objects even if not strictly needed to make it easier in the future to turn on the pedantic ``-Weffc++`` flag
```suggestion
    std::vector<int> anBandList{};
    std::vector<double> adfBurnValues{};
    bool bInverse = false;
    std::string osFormat{};
    bool b3D = false;
    GDALProgressFunc pfnProgress = GDALDummyProgress;
    void *pProgressData = nullptr;
    std::vector<std::string> aosLayers{};
    std::string osSQL{};
    std::string osDialect{};
    std::string osBurnAttribute{};
    std::string osWHERE{};
    CPLStringList aosRasterizeOptions{};
    CPLStringList aosTO{};
    double dfXRes = 0;
    double dfYRes = 0;
    CPLStringList aosCreationOptions{};
    GDALDataType eOutputType = GDT_Undefined;
    std::vector<double> adfInitVals{};
    std::string osNoData{};
    OGREnvelope sEnvelop{};
    int nXSize = 0;
    int nYSize = 0;
    OGRSpatialReference oOutputSRS{};
```